overlord/hookstate/ctlcmd/get.go

```diff
@@ -338,3 +357,92 @@ func (c *getCommand) getInterfaceSetting(context *hookstate.Context, plugOrSlot
 		return nil, false, err
 	})
 }
+
+func (c *getCommand) getRegistryView(ctx *hookstate.Context, account, registryName, viewName string) error {
+	ctx.Lock()
+	defer ctx.Unlock()
+
+	registryView := registryName + "/" + viewName
+	if ok, err := hasConnectedRegistryPlug(ctx, account, registryView); err != nil {
+		return err
+	} else if !ok {
+		return fmt.Errorf(i18n.G("cannot get %s/%s: cannot find connected plug for %s"), account, registryView, ctx.InstanceName())
+	}
+
+	registryAssert, err := assertstate.Registry(ctx.State(), account, registryName)
+	if err != nil {
+		if errors.Is(err, &asserts.NotFoundError{}) {
+			return fmt.Errorf(i18n.G("cannot get %s/%s: registry not found"), account, registryView)
+		}
+		return err
+	}
+	reg := registryAssert.Registry()
+
+	view := reg.View(viewName)
+	if view == nil {
+		return fmt.Errorf(i18n.G("cannot get %s/%s: view not found"), account, registryView)
+	}
+
+	tx, err := registrystate.RegistryTransaction(ctx, reg)
+	if err != nil {
+		return err
+	}
+
+	res, err := registrystate.GetViaViewInTx(tx, view, c.Positional.Keys)
+	if err != nil {
+		return err
+	}
+
+	return c.printPatch(res)
+}
+
+func hasConnectedRegistryPlug(ctx *hookstate.Context, account, registryView string) (bool, error) {
+	repo := ifacerepo.Get(ctx.State())
+	plugs := repo.Plugs(ctx.InstanceName())
+
+	for _, plug := range plugs {
+		if plug.Interface != "registry" {
```

Comment:
This interface doesn't exist yet, right?

Reply:
There's a PR up for it https://github.com/snapcore/snapd/pull/14113